app/views/posts/_cards.html.erb

```diff
@@ -2,38 +2,49 @@
 
 <div class="posts">
   <% @posts.each do |post| %>
-    <%= link_to post_path(post.id) do%>
-      <div class="post">
+    <div class="post">
+      <%= link_to post_path(post.id) do%>
         <div class="post-header">
           <div class="user">
             <%= image_tag('avatar-default.webp') %>
             <div>
-              <h2><%=post.user.name%> <%=post.user.lastName%></h2>
+              <h2><%= post.user.name %> <%= post.user.lastName %></h2>
               <p><%= post.created_at.strftime("%d-%m-%Y a las %H:%M") %></p>
             </div>
           </div>
           <i class="bi bi-three-dots"></i>
         </div>
-        <div class="post-body">
-          <%=post.body%>
-        </div>
-        <div class="post-images">
-          <% if post.images.attached? %>
-            <% post.images.each do |image| %>
-              <div>
-                <%= image_tag image.representation(resize_to_limit: [300, 300]) %>
-              </div>
-            <% end %>
+      <% end %>
+      <div class="post-body">
+        <%= post.body %>
+      </div>
+      <div class="post-images">
+        <% if post.images.attached? %>
+          <% post.images.each do |image| %>
+            <div>
+              <%= image_tag image.representation(resize_to_limit: [300, 300]) %>
+            </div>
           <% end %>
-        </div>
-        <div class="post-footer">
-          <div class="actions">
-            <i class="bi bi-heart-fill"></i>
-            <i class="bi bi-chat-fill"></i>
-            <i class="bi bi-share-fill"></i>
-          </div>
+        <% end %>
+      </div>
+      <div class="post-footer">
+        <div class="actions">
+          <i class="bi bi-heart-fill"></i>
+          <i class="bi bi-chat-fill"></i>
+          <i class="bi bi-share-fill"></i>
         </div>
       </div>
-    <%end%>
+    </div>
+  <% if @user_likes[post.id] %>
+    <%= form_with(model: @user_likes[post.id], url: post_like_path(post, @user_likes[post.id]), method: :delete) do |form| %>
+      <%= form.hidden_field :id %>
+      <%= form.submit "Unlike", class: "like-button" %>
+    <% end %>
+  <% else %>
+    <%= form_with(model: @like, url: post_likes_path(post), method: :post) do |form| %>
+      <%= form.submit "Like", class: "like-button" %>
+    <% end %>
   <% end %>
```

Comment:
If you use turbo streams, you must have to wrap this piece of code into a turbo_frame_tag.

```ruby
<%= turbo_frame_tag 'likes' do %>
  <% if @user_likes[post.id] %>
    <%= form_with(model: @user_likes[post.id], url: post_like_path(post, @user_likes[post.id]), method: :delete) do |form| %>
      <%= form.hidden_field :id %>
      <%= form.submit "Unlike", class: "like-button" %>
    <% end %>
  <% else %>
    <%= form_with(model: @like, url: post_likes_path(post), method: :post) do |form| %>
      <%= form.submit "Like", class: "like-button" %>
    <% end %>
  <% end %>
<% end %>
```

I suggest you separate the like and unlike button into partials and just render which you need based on the if statement